src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerMiddleware.cs

```diff
@@ -118,15 +118,23 @@ private async Task HandleException(HttpContext context, ExceptionDispatchInfo ed
             }
             try
             {
-                ClearHttpContext(context);
-
                 var exceptionHandlerFeature = new ExceptionHandlerFeature()
                 {
                     Error = edi.SourceException,
                     Path = originalPath.Value!,
+                    Endpoint = context.GetEndpoint(),
+                    RouteValues = context.Features.Get<IRouteValuesFeature>()?.RouteValues
                 };
+
+                ClearHttpContext(context);
+
                 context.Features.Set<IExceptionHandlerFeature>(exceptionHandlerFeature);
                 context.Features.Set<IExceptionHandlerPathFeature>(exceptionHandlerFeature);
+                if (exceptionHandlerFeature.Endpoint is not null || exceptionHandlerFeature.RouteValues is not null)
+                {
+                    context.Features.Set<IExceptionHandlerEndpointFeature>(exceptionHandlerFeature);
+                }
```

Comment:
Should this feature always be set? The feature always present but with null values seems preferable. I think we had this debate about IActivityFeature.

Reply:
I guess it's simpler to always set it. The consumer will always have to null check when retrieving the feature regardless. E.g. someone could navigate directly to the /error page without an exception and they wouldn't have any of these features set.